### PR TITLE
Do not split call of scss function with leading dash

### PR DESCRIPTION
**Pull request by auvred**
## Description

Fixes https://github.com/prettier/prettier/issues/15369

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-css/print/comma-separated-value-group.js [hunk 1]
```diff
@@ -269,7 +269,8 @@ function printCommaSeparatedValueGroup(path, options, print) {
       options.parser === "scss" &&
       isMathOperator &&
       iNode.value === "-" &&
-      iNextNode.type === "value-func"
+      iNextNode.type === "value-func" &&
+      locEnd(iNode) !== locStart(iNextNode)
```
Comment (fisker) [inline src/language-css/print/comma-separated-value-group.js:273]: Is this `-foo()` syntax only supported by SCSS? I don't think LESS support it. So, we should check `options.parser` here.

Comment (auvred) [reply src/language-css/print/comma-separated-value-group.js]: Thanks, updated it

**Comment by fisker** [comment]
@auvred I guess there is no way to tell

```css
width: -double(1em);

// Mean
width: -(double(1em));

// Or
width: (-double)(1em);
```

So I guess we can use your logic to check if there was space there.
Maybe we can remove that part added in #15129?

**Comment by fisker** [comment]
Sorry for not noticing this PR and #15129 are changing the same part. We are good..

File: src/language-css/print/comma-separated-value-group.js [hunk 2]
```diff
@@ -269,7 +269,8 @@ function printCommaSeparatedValueGroup(path, options, print) {
       options.parser === "scss" &&
       isMathOperator &&
       iNode.value === "-" &&
-      iNextNode.type === "value-func"
+      iNextNode.type === "value-func" &&
+      (options.parser !== "scss" || locEnd(iNode) !== locStart(iNextNode))
```
Comment (fisker) [inline src/language-css/print/comma-separated-value-group.js:273]: Didn't notice the parser already checked above.

Comment (auvred) [reply src/language-css/print/comma-separated-value-group.js:273]: Ohh.. I missed that too

File: tests/format/scss/scss/scss.scss [hunk 3]
```diff
@@ -1090,6 +1090,16 @@ div {
   margin: - pow(2, 2) * 100px;
 }
 
+div {
+  margin: -double(-double(1em));
```
Comment (fisker) [inline tests/format/scss/scss/scss.scss:1094]:
```suggestion
  margin: -double(-double(1em));
  margin: -double(- double(1em));
  margin: - double(-double(1em));
```

Comment (auvred) [reply tests/format/scss/scss/scss.scss:1094]: https://github.com/prettier/prettier/pull/15370/commits/fed17b6174a4efd3e4554a2811880fda38147527

**Review by fisker — APPROVED** [review]
LGTM, thank you!

File: tests/format/scss/scss/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -4835,6 +4869,16 @@ div {
...
 }
 
+div {
+  margin: -double(-double(1em));
+  margin: -double(- double(1em));
+  margin: - double(-double(1em));
+}
+
+@function asdf() {
+  @return (width: -double(1em) ;);
```
Comment (fisker) [inline tests/format/scss/scss/__snapshots__/jsfmt.spec.js.snap:4879]: This looks like a bug to me too.

File: tests/format/scss/scss/scss.scss [hunk 5]
```diff
@@ -1090,6 +1090,18 @@ div {
...
+  margin: -double(- double(1em));
+  margin: - double(-double(1em));
+}
+
+@function asdf() {
+  @return (
+    width: -double(1em);
+  )
+}
```
Comment (fisker) [inline tests/format/scss/scss/scss.scss:1103]:
```suggestion
@function asdf() {
  @return width: -double(1em);
}
```

Comment (auvred) [reply tests/format/scss/scss/scss.scss]: Changed it https://github.com/prettier/prettier/pull/15370/commits/3f9921e94db056236ff5a32c583ce37023ae698d

**Review by fisker — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**